src/components/members/filters/RangeModal.js

```diff
@@ -0,0 +1,97 @@
+import styles from "./modals.module.scss";
+import {Button, Dai, Modal, ModalOverlay} from "@unioncredit/ui";
+import {SingleInputControl} from "./SingleInputControl";
+import {useEffect, useState} from "react";
+import {Range} from "../../../constants/filters";
+import {RangeInputControl} from "./RangeInputControl";
+import {ReactComponent as FilterIcon} from "images/filter.svg";
+
+const initialValuesState = {
+  [Range.LTE]: "",
+  [Range.GTE]: "",
+  [Range.BETWEEN]: "",
+};
+
+export const RangeModal = ({id, open, title, isDai, filters, handleClose}) => {
+  const [selected, setSelected] = useState("");
+  const [values, setValues] = useState(initialValuesState);
+
+  useEffect(() => {
+    const item = filters.rangeValues.find(v => v.key === id);
+    setSelected(item ? item.selected : "");
+    setValues(item ? item.values : initialValuesState);
+  }, [filters.rangeValues]);
+
+  useEffect(() => {
+    setValues(initialValuesState);
+  }, [selected]);
```

Comment:
This doesn't need to be here because it is not a dependency of the useEffect callback.